File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let days = [];
    for(let i = 0;i<progresses.length;i++){
        const day = Math.ceil((100 - progresses[i])/speeds[i]);
        days.push(day);
    }
    let p = 0;
    let pp = 0;
    let cnt =0;
    console.log(days);
    while(p !== days.length){
        if(days[p]>=days[pp]){
            cnt++;
            pp++;
        }
        else{
            answer.push(cnt);
            cnt=0;
            p=pp;
        }
    }
    return answer;
}
/*


*/

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
#include <math.h>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> days;
    for(int i = 0;i<progresses.size();i++){
        int day = ceil((100.0-progresses[i])/speeds[i]);
        days.push_back(day);
    }
    int p = 0;
    int pp = 0;
    int cnt = 0;
    while(p != days.size()){
        if(days[p]>=days[pp]){
            cnt++;
            pp++;
        }
        else{
            answer.push_back(cnt);
            cnt=0;
            p=pp;
        }
        if(pp==days.size()){
            answer.push_back(cnt);
            break;
        };
    }
    return answer;
}